examples/loki/nurf_data.py

```diff
@@ -102,120 +102,173 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
+        uv_ik_ref=4*np.ones((1,uv_nb_ref))  #new image key: 4 for reference
+        #print(ik_ref)
+        
+        # assmebling of image_key
+        uv_image_key=np.column_stack((uv_ik_spectra,uv_ik_dark, uv_ik_ref))  #all_data is as well organised in columns
         
-        #print(list(hf['/entry/'].keys()))
         
-        # create the various subgrous and their attributes
-
         # UV subgroup
         grp_uv = hf.create_group("/entry/instrument/uv")
-        grp_uv.attrs["NX_class"] = 'NXdetector'
-        grp_uv.create_dataset('description', data='UV')
-        grp_uv.create_dataset('type',
-                              data='HR4PRO-UV-VIS-ES, DH-2000-FHS-DUV-TTL, QP600-2-SR')
-    
-        # create the various datasets for UV
-        uv_inttime = grp_uv.create_dataset('uv_integration_time',
-                                           data=data['UV_IntegrationTime'],
-                                           shape=data[
-                                               'UV_IntegrationTime'].shape,
-                                           dtype=np.int32)
-        uv_inttime.attrs['long_name'] = 'uv_integration_time'
-        uv_inttime.attrs['units'] = 's'  # TODO: unit to be verified
-
-        uv_bkg = grp_uv.create_dataset('uv_background',
-                                       data=data['UV_background'],
-                                       shape=data['UV_background'].shape,
-                                       dtype=np.float32)
-        uv_bkg.attrs['long name'] = 'uv_background'
-
-        uv_int0 = grp_uv.create_dataset('uv_intensity0',
-                                        data=data['UV_intensity0'],
-                                        shape=data['UV_intensity0'].shape,
-                                        dtype=np.float32)
-        uv_int0.attrs['long name'] = 'uv_intensity0'
-
-        uv_spectra = grp_uv.create_dataset('uv_spectra',
-                                           data=data['UV_spectra'],
-                                           shape=data['UV_spectra'].shape,
-                                           dtype=np.float32)
-        uv_spectra.attrs['long name'] = 'uv_spectra'
-
-        uv_wavelength = grp_uv.create_dataset('uv_wavelength',
-                                              data=data['UV_wavelength'],
+        grp_uv.attrs["NX_class"] = 'NXdata'
+        
+        # subgroup for uv all data (sample, dark, reference)
+        uv_signal=grp_uv.create_group("uv_all_data")
+        uv_signal_data=uv_signal.create_dataset('data', data=uv_all_data,
+                                           shape=uv_all_data.shape)
+        uv_signal_data.attrs['long_name']= 'uv_all_data'
+        uv_signal_data.attrs['units']= ''
+        uv_signal_data.attrs['signal']= 'data'  #indicate that the main signal is data 
+        uv_signal_data.attrs['axes']= [ "wavelength", "." ]   #see example in NXdata, time as x-axis is first entry
```

Comment:
The `signal` and `axes` attributes of the "data" field are (to my understanding) deprecated. Use `signal` and `axes` attributes of the *group*.

Reply:
If `all_uv_data` becomes just a dataset, then I could move the attributes to the `uv` group. 
For example:
```
grp_uv.attrs['signal']= 'data'  #indicate that the main signal is data 
grp_uv.attrs['axes']= [ "wavelength", "." ]
```
Would this be what you suggest?
I am unsure about the order of the `axes` parameters. In the NXData manual it says `@axes=["time", "pressure"]` for a plot where pressure varies over time.

Reply:
> I am unsure about the order of the axes parameters

Same order as the axes in your numpy array of values. For what you implemented currently using `numpy.column_stack` this would indeed be what you wrote, but if you follow my suggestion to keep wavelength as the "fast" dimension then it should be

```python
grp_uv.attrs['axes']= [".", "wavelength"]
```